src/aws-location-mcp-server/tests/test_server.py

```diff
@@ -0,0 +1,142 @@
+    """Test the get_coordinates tool."""
+    # Set up test data
+    location = "Seattle"
+    
+    # Call the function
+    with patch('awslabs.aws_location_server.server.location_client.location_client', mock_boto3_client):
+        result = await get_coordinates(mock_context, location=location)
+    
+    # Verify the result
+    assert result['location'] == location
+    assert result['formatted_address'] == 'Seattle, WA, USA'
+    assert result['coordinates']['longitude'] == -122.3321
+    assert result['coordinates']['latitude'] == 47.6062
+    assert result['country'] == 'USA'
+    assert result['region'] == 'Washington'
+    assert result['municipality'] == 'Seattle'
+    
+    # Verify the boto3 client was called correctly
+    mock_boto3_client.search_place_index_for_text.assert_called_once_with(
+        IndexName='ExamplePlaceIndex',
+        Text=location,
+        MaxResults=1
+    )
+
+
+@pytest.mark.asyncio
+async def test_search_places_error(mock_context):
+    """Test the search_places tool with an error."""
+    # Set up test data
+    query = "Seattle"
+    
+    # Mock boto3 client to raise an exception
+    mock_client = MagicMock()
+    error_response = {'Error': {'Code': 'ResourceNotFoundException', 'Message': 'Place index not found'}}
+    mock_client.search_place_index_for_text.side_effect = botocore.exceptions.ClientError(
+        error_response, 'SearchPlaceIndexForText'
+    )
+    
+    # Call the function
+    with patch('awslabs.aws_location_server.server.location_client.location_client', mock_client):
+        result = await search_places(mock_context, query=query)
+    
+    # Verify the result contains an error
+    assert 'error' in result
+    assert 'place index not found' in result['error'].lower()
+    
+    # Verify the context.error was called
+    mock_context.error.assert_called_once()
+
+
+@pytest.mark.asyncio
+async def test_get_coordinates_no_results(mock_context):
+    """Test the get_coordinates tool with no results."""
+    # Set up test data
+    location = "NonexistentPlace"
+    
+    # Mock boto3 client to return no results
+    mock_client = MagicMock()
+    mock_client.search_place_index_for_text.return_value = {'Results': []}
+    
+    # Call the function
+    with patch('awslabs.aws_location_server.server.location_client.location_client', mock_client):
+        result = await get_coordinates(mock_context, location=location)
+    
+    # Verify the result contains an error
+    assert 'error' in result
+    assert 'no results found' in result['error'].lower()
+    
+    # Verify the context.error was called
+    mock_context.error.assert_called_once()
+
+
+def test_location_client_initialization():
+    """Test the LocationClient initialization."""
+    # Test with environment variables
+    with patch.dict(os.environ, {
+        'AWS_REGION': 'us-west-2',
+        'AWS_ACCESS_KEY_ID': 'test-key',
+        'AWS_SECRET_ACCESS_KEY': 'test-secret',
```

Comment:
Consider using ficticious values:
```suggestion
        'AWS_REGION': 'us-west-2',
        'AWS_ACCESS_KEY_ID: 'ASIAIOSFODNN7EXAMPLE'
        'AWS_SECRET_ACCESS_KEY': 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY'
        'AWS_SESSION_TOKEN': '=AQoEXAMPLEH4aoAH0gNCAPy...truncated...zrkuWJOgQs8IZZaIv2BXIa2R4Olgk'
```